### PR TITLE
Rust-based test suite

### PR DESCRIPTION
**Pull request by tarcieri**
NOTE: Tests are failing due to segfaults and are commented out!

Leverages code from the curve25519-dalek crate to test libristretto255

Testing Linux on Travis CI (OS X and Windows are presently build-only)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,9 +24,19 @@ https://ristretto.group/
...
 This library is in an extremely early stage of development and is not ready to
-be used yet.
+be used yet. Check back later for updates.
 
-Check back later for updates.
+### Known Issues
+
+#### Travis CI
+
+* Sporadic SEGV on Travis C with clang when running `ristretto_gen_tables`, e.g.
```
Comment (tarcieri) [inline README.md]: 😭

Comment (hdevalence) [reply README.md]: seems ... less than ideal

Comment (tarcieri) [reply README.md]: It gets worse:

https://github.com/Ristretto/libristretto255/pull/17/files#diff-be5c31b26c462f22ad555004082e6eaaR433

**Comment by tarcieri** [comment]
Merged this despite the known SEGVs to get a basic test framework in place. Will open issues for the SEGVs.

**Pull request merged**